Added a new article Known compatibility issues with Android apps (#83)
AdguardTeam/KnowledgeBase: docs/adguard-for-android/solving-problems/compatibility-issues.md

---
title: Known compatibility issues with Android apps
sidebar_position: 14
---

### VPN apps

Other VPN apps are incompatible with AdGuard running in the Local VPN filtering mode since you cannot have two active VPNs at the same time. To solve this problem we suggest you:

> * use AdGuard VPN — it has Integrated mode that allows two apps to operate seamlessly

Reviewer: Where's a link to AdGuard VPN?

* use AdGuard in Local HTTP Proxy mode. *Please note that 'manual' HTTP Proxy mode is not supported on devices since Android 10*
* if you use a third-party VPN app, figure out whether it supports SOCKS4/SOCKS5 mode. If it does, you can configure your VPN app to work as outbound proxy and set up local outbound proxy with params from the third-party app. For instance, AdGuard can be automatically configured with 'Orbot: Proxy with Tor'

### Private DNS

> The Private DNS feature was introduced in Android Pie. Before version Q, Private DNS didn't break our logic and the DNS forwarding through AdGuard worked normally. But starting from version Q, the presence of Private DNS forces apps to redirect traffic through the system resolver instead of AdGuard. See Android [devs blog](https://android-developers.googleblog.com/2018/04/dns-over-tls-support-in-android-p.html) for more details.

Reviewer 2: It is still 'our' logic instead. Maybe we can use 'AdGuard DNS filtering logic' or something?

> Some device manufacturers keep Private DNS settings hidden and set 'Automatic' mode as a default one. Thus, disabling Private DNS is impossible but we can make the system think that the upstream is not valid. The system will use AdGuard as a bootstrap for the first host resolution and we can block it with a `$network` rule. A failed bootstrap request causes the system to redirect DNS traffic through AdGuard.

### Unsupported browsers

| **App name** | **Package name** | **Filtering mode** |
| :----: | :----: | ---: | ---: |
| Opera mini      | com.opera.mini.android | All |
| Opera mini with Yandex | com.opera.mini.android.yandex | All |

> | UC Browser | com.UCMobile.intl | All|

Reviewer: UC Browsers have no issues *by default*, it's limited only when data compression is enabled.

Generally, the content plan:
1. Browser name
2. Incompatibility description
3. Solution (if any)

For instance, for all UC-family browsers we can have one section:

```
#### UC Browser

By default, AdGuard filters UC Browser just okay. However, if you enable "data compression"  in that browser, AdGuard won't be able to apply HTTPS filtering.

**Solution**

The only solution is to not enable data compression in UC Browser.

#### Opera mini

Opera mini drives traffic through a compression proxy by default and AdGuard is not able to decompress and filter it at the same time.

**Solution**

There is no solution at this moment.
```

The browsers need to be sorted by popularity.
UC > Opera > Dolfin > Puffin

---
title: Known compatibility issues with Android apps
sidebar_position: 14
---

> ## Conflict apps

Reviewer 3: ```suggestion
## Conflicting apps and features
```

> ## Conflicting apps and features

Reviewer: Why is the title different from `Known compatibility issues with Android apps`

Reviewer: Moreover, why do we need this header here? The page already has an H1 header.

> ### Various VPN apps

Reviewer 3: VPN apps

> All VPN apps are incompatible with AdGuard running in VPN filtering mode since you cannot have two active VPNs at the same time. To solve this problem we can suggest you:

Reviewer 3: ```suggestion
Other VPN apps are incompatible with AdGuard running in the Local VPN filtering mode since you cannot have two active VPNs at the same time. To solve this problem we suggest you:
```

> * to use AdGuard VPN that has a compatibility mode with AdGuard out-of-the-box

Reviewer 3: * use AdGuard VPN that is compatible with AdGuard "out-of-the-box"
/
* use AdGuard VPN — it has Integrated mode that allows two apps to operate seamlessly

> * to use AdGuard in Local HTTP Proxy mode. *Please note that 'manual' HTTP Proxy mode is broken since Android 10*

Reviewer 3: что значит broken? мб написать не поддерживается на устройствах, начиная с андроид 10?

Reviewer 3: to тоже убрать можно

> * to figure out if the third-party VPN app supports SOCKS4/SOCKS5 mode. In this case you can configure a VPN app to work as outbound proxy and to set up local-host outbound proxy with params from the third-party app. AdGuard can be automatically configured with 'Orbot: Proxy with Tor'

Reviewer 3: * if you use a third-party VPN app, figure out whether it supports SOCKS4/SOCKS5 mode. If it does, you can configure your VPN app to work as outbound proxy and set up local outbound proxy with params from the third-party app. For instance, AdGuard can be automatically configured with 'Orbot: Proxy with Tor'

> ### Samsung Pay

Reviewer 2: Samsung Pay is not a 'conflict app' since it doesn't brake AdGuard's flow. On the contrary, it is violated by AdGuard

Reviewer 3: to my mind, if we call apps conflicting it means that they are incompatible and somehow hinder each other's operation
in the case of Samsung Pay, it's AdGuard that violates it

Reviewer: We have a separate article about Samsung Pay, remove this section

The Samsung Pay app does not work with AdGuard due to the presence of an active VPN interface. This issue is region-specific and all known issues were reported by users from South Korea. Find more details [here](https://github.com/AdguardTeam/AdguardForAndroid/issues/1209).

> ### Private DNS

Reviewer 2: Private DNS is not an 'app', we should move it in a separate section

Reviewer 3: agree, renamed the above section as ## Conflicting apps and features

> The Private DNS feature was introduced in Android Pie. However, before version Q the private DNS didn't break our logic, and the DNS forwarding through AdGuard worked normally. But since version Q, the presence of a private DNS forces apps to redirect traffic through the system resolver instead of AdGuard. See Android [devs blog](https://android-developers.googleblog.com/2018/04/dns-over-tls-support-in-android-p.html) for more details.

Reviewer 2: I think that we should replace 'us'/'our' with 'AdGuard'

> The Private DNS feature was introduced in Android Pie. However, before version Q the private DNS didn't break our logic, and the DNS forwarding through AdGuard worked normally. But since version Q, the presence of a private DNS forces apps to redirect traffic through the system resolver instead of AdGuard. See Android [devs blog](https://android-developers.googleblog.com/2018/04/dns-over-tls-support-in-android-p.html) for more details.

Reviewer 3: ```suggestion
The Private DNS feature was introduced in Android Pie. Before version Q, Private DNS didn't break our logic and the DNS forwarding through AdGuard worked normally. But starting from version Q, the presence of Private DNS forces apps to redirect traffic through the system resolver instead of AdGuard. See Android [devs blog](https://android-developers.googleblog.com/2018/04/dns-over-tls-support-in-android-p.html) for more details.
```

> > Some device manufacturers keep Private DNS settings hidden and set 'Automatic' mode as a default one. Thus, disabling the private DNS is impossible, but we can make the system think that the upstream is not valid. The system will use us as a bootstrap for the first host resolution and we can block it with $network rule. A failed bootstrap request causes the system to redirect DNS traffic through AdGuard.

Reviewer 3: ```suggestion
> Some device manufacturers keep Private DNS settings hidden and set 'Automatic' mode as a default one. Thus, disabling Private DNS is impossible but we can make the system think that the upstream is not valid. The system will use AdGuard as a bootstrap for the first host resolution and we can block it with a `$network` rule. A failed bootstrap request causes the system to redirect DNS traffic through AdGuard.
```

* use [AdGuard VPN](https://adguard-vpn.com/ro/welcome.html) — it has Integrated mode that allows two apps to operate seamlessly
* use AdGuard in Local HTTP Proxy mode. *Please note that 'manual' HTTP Proxy mode is not supported on devices since Android 10*
* if you use a third-party VPN app, figure out whether it supports SOCKS4/SOCKS5 mode. If it does, you can configure your VPN app to work as outbound proxy and set up local outbound proxy with params from the third-party app. For instance, AdGuard can be automatically configured with 'Orbot: Proxy with Tor'

### Private DNS

The Private DNS feature was introduced in Android Pie. Before version Q, Private DNS didn't break AdGuard DNS filtering logic and the DNS forwarding through AdGuard worked normally. But starting from version Q, the presence of Private DNS forces apps to redirect traffic through the system resolver instead of AdGuard. See Android [devs blog](https://android-developers.googleblog.com/2018/04/dns-over-tls-support-in-android-p.html) for more details.

> > Some device manufacturers keep Private DNS settings hidden and set 'Automatic' mode as a default one. Thus, disabling Private DNS is impossible but we can make the system think that the upstream is not valid by blocking it with a `$network` rule. For instance, if the system uses Google DNS by default, we can add rules `|8.8.4.4^$network` and `|8.8.8.8^$network` to block Google DNS.

Reviewer 2: I think we should use rule with port `853` as an example to not block all Google IP addresses.

Reviewer 2: @ameshkov what do you think about this? Or, as an example, leave it without a port?

> > Some device manufacturers keep Private DNS settings hidden and set 'Automatic' mode as a default one. Thus, disabling Private DNS is impossible but we can make the system think that the upstream is not valid. The system will use AdGuard as a bootstrap for the first host resolution and we can block it with a `$network` rule. A failed bootstrap request causes the system to redirect DNS traffic through AdGuard.

Reviewer: ```suggestion
> Some device manufacturers keep Private DNS settings hidden and set 'Automatic' mode as a default one. Thus, disabling Private DNS is impossible but we can make the system think that the upstream is not valid by blocking it with a `$network` rule. For instance, if the system uses Google DNS by default, we can add rules `|8.8.4.4^$network` and `|8.8.8.8^$network` to block Google DNS.
```

### Unsupported browsers

1. **UC Browsers**: UC Browser, UC Browser for x86, UC Mini, UC Browser HD

> To be able to filter HTTPS traffic, AdGuard needs to install a certificate into the device's user storage. UC-family browsers don't trust user certificates, so AdGuard cannot perform HTTPS filtering there. 

Reviewer: ```suggestion
To be able to filter HTTPS traffic, AdGuard requires the user to install a certificate into the device's trusted user certificates. Unfortunately, UC-family browsers don't trust user certificates, so AdGuard cannot perform HTTPS filtering there.
```

**Solution**

> To solve this problem, move the certificate to the system storage. This will require granting root access. 

Reviewer: ```suggestion
**Requires root access*. To solve this problem, move the certificate to the system certificates store.
```

2. **Opera mini**: Opera mini, Opera mini with Yandex

Opera mini drives traffic through a compression proxy by default and AdGuard is not able to decompress and filter it at the same time.

**Solution**

There is no solution at this moment.

3. **Dolphin Browser**: Dolphin Browser, Dolphin Browser Express

AdGuard cannot filter its traffic when operating in **Manual proxy mode** because this browser ignores system proxy settings.

**Solution**

> There is no solution at this moment.

Reviewer: ```suggestion
Use *Local VPN* filtering mode.
```
